### ISSUE TITLE
Error during parsing with custom row delimiter

### ISSUE THREAD

**Issue by KrzychuJedi** [adaltas/node-csv-parse#211]
I have file with such content:

```
    'Timestamp;Event1;Event2;PM;F;V;U\r\n' +
    '2018-09-13 00:00:00.000;1;1;418;0.00;32.50;190;\r\n' +
    '2018-09-13 00:01:00.000;1;0;395;0.00;32.75;0;\r\n' +
```

It is not standard CSV file because after last column is extra delimiter. So I set custom row delimiter to match ';\r\n' this way:

`    parser(
      csvLog.fileContent,
      {
        delimiter: ;,
        rowDelimiter: [';\r\n', '\r\n', ';\n', '\n', '\r']
      },
      (err: Error, output: any) => this.onLogParsed(err, output, csvLog)
    )` 

Result is:

`[
  [
    "Timestamp",
    "Event1 (DM3)",
    "Event2 (DM5)",
    "PM [Unit] (AM0)",
    "FreqOut1 [Hz] (AM1)",
    "TempVar [C] (AM3)",
    "PM2 [Unit] (AM4)"
  ],
  [
    "2018-09-13 00:00:00.000",
    "1",
    "1",
    "418",
    "0.00",
    "32.50",
    "190"
  ],
  [
    "018-09-13 00:01:00.000",
    "1",
    "0",
    "395",
    "0.00",
    "32.75",
    "0"
  ]
]`

Parser removes first sign from third and all next lines.

Test: https://runkit.com/krzychujedi/csv-parse-rowdelimiter-error

**Comment by wdavidw** [comment]
In your example, you have a different number of fields per line, you could use the `relax_column_count`:

```coffee
input = 'Timestamp;Event1;Event2;PM;F;V;U\r\n' +
  '2018-09-13 00:00:00.000;1;1;418;0.00;32.50;190;\r\n' +
  '2018-09-13 00:01:00.000;1;0;395;0.00;32.75;0;\r\n'
parse input, delimiter: ';', relax_column_count: true, (err, data) ->
  console.log err, data
```

**Comment by KrzychuJedi** [comment]
As I mention in first message it's not standard CSV file - it has extra delimiter on the end of line but it shouldn't be treated like additional column so I set as row delimiter ';\r\n'

In this case this solution gives me 7 items in header row and 7 items in each next row but it loose data (first sign of each third and next line).

If I use relax_column_count: true It will gives me 8 values where last one will be empty - I don't want that.

**Comment by wdavidw** [comment]
OK, i get it, output on the 2nd record is expected to be `2018-09-13 00:01:00.000` but is in fact truncated to `018-09-13 00:01:00.000`. That is not a feature, it is surely a bug.

**Comment by wdavidw** [comment]
The new version 3.1.3 shall fix the issue with the commit 41b05bf84500c796298b017d95706bd1a6d52974

**Comment by KrzychuJedi** [comment]
It's working. Thanks for quick reaction.
